### DISCUSSION TITLE
Tooling Übersicht

### DISCUSSION THREAD

**Discussion by Kurpanik** [globallightingdata/gldf#17] [General]
## Erstellung von LDF-Containern

- **Name des Tools**
  - ?
- **Zweck/Aufgaben**
  - Container erstellen, Daten validieren
  - Siehe Issue [#103: Containerformat definieren und dokumentieren](https://github.com/Dx-Rx/LDF/issues/103)
- **Wann benötigt**
  - Vermutlich zum Release
- **Entwicklung gemeinsam/getrennt**
  - ?
- **Link zum Repository**
  - Keins vorhanden

## Erstellung von 3D Modellen

- **Name des Tools**
  - ?
- **Zweck/Aufgaben**
  - 3D Oberfläche, Erstellung von Modellen
  - Import und Ausrichten von .obj Dateien
  - Definieren von LEOs, Positionen und Rotationen
  - Definieren von Gelenken, Freiheitsgraden, Standardwerten
  - Verpacken der .obj Modelle und .xml mit Lichtaustritts- und Gelenkinformationen zu einem Container
- **Wann benötigt**
  - Vermutlich zum Release
- **Entwicklung gemeinsam/getrennt**
  - ?
- **Link zum Repository**
  - Keins vorhanden
  - (Gemeinsames Geometry-XSD: https://github.com/Dx-Rx/Geometry)

## XML Parser

- **Name des Tools**
  - ?
- **Zweck/Aufgaben**  
  - Lesen der Leuchtendaten- und Geometrie-Container
  - Parsen des XML Leuchtendatenformats
  - Parsen des XML Geometriedatenformats
- **Wann benötigt**
  - Sobald DIALux/Relux den Import unterstützen sollen
- **Entwicklung gemeinsam/getrennt**
  - Vermutlich getrennt?
- **Link zum Repository**
  - Keins vorhanden

## Prüfsummen-Dienst

- **Name des Tools**
  - ?
- **Zweck/Aufgaben**  
  - Prüfsumme für das `Checksum`-Attribut erstellen
  - `Checksum`-Attribut für ein XML prüfen
  - `Checksum`-Datei in einem Container prüfen
- **Wann benötigt**
  - ?
- **Entwicklung gemeinsam/getrennt**
  - ?
  - Wenn getrennt, muss die Spezifikation für die Erstellung/Verifikation kompatibel sein
- **Link zum Repository**
  - Keins vorhanden
  - Issue [#62: Prüfsumme technisch klären](https://github.com/Dx-Rx/LDF/issues/62)

## Merkmalsserver

- **Name des Tools**
  - ?
- **Zweck/Aufgaben**
  - ?
- **Zum Release benötigt**
  - ?
- **Entwicklung gemeinsam/getrennt**
  - ?
- **Link zum Repository**
  - Keins vorhanden

**Comment by robertrelux** [comment]
Danke für diese Übersicht und diesen Diskussionsbereich.
Ein paar kleine Anmerkungen von mir:
- in "Entwicklung" ist ein Rechtschreibfehler (i->u)
- die Tool Frage für einen LDF (zip) Container ist für mich neu, aber deshalb nicht weniger wichtig; ich war davon ausgegangen, dass wir das nur definieren und dokumentieren müssen und kein Tool nötog ist. Die Erzeugen können doch selber die XML und die Assets zippen und umbennen. Übersehe ich etwas?
- Das Geometrie Tool würde zusammen Sinn machen. Hier würde ich noch die Frage nach weiteren Tools hinzufügen (Desktop Tool als Ergänzung zum Online Tool oder Autodesk PlugIns)
- den Parser sehe ich auch eher getrennt ... unsere internen Schnittstellen dürften zu verschieden sein
- den Prüfsummendienst sollten wir gemeinsam machen, zumindest die Kern API, ev. können wir von unseren Webseiten die API nutzen oder sogar auf eine gemeinsame Seite verlinken.
- der Merkmalsserver ist auch für mich ein großes Fragezeichen ... 
Nach meinen Infos wollen Markus und Dieter in den ersten Januar Wochen noch mal beraten, wie weit die Zusammenarbeit in Tools gehen soll.

**Comment by Kurpanik** [comment]
Hi Robert, danke für dein Feedback.

Zum Tool für das ZIP: Als ich mich mit Fragestellungen zum Container beschäftigt hab, fielen mir immer mehr Dinge ein, die ein Tool hier leisten könnte. Ich hab sie in diesem Issue hier versucht detaillierter zu spezifizieren: https://github.com/Dx-Rx/LDF/issues/103. Und um sich das ganze etwas besser vorstellen zu können, eine Oberfläche dort skizziert. 

Können das ja gern mal in der Gruppe diskutieren - vielleicht ist es auch unnötig, wenn die Hersteller diese Dinge ohne Hilfsmittel umsetzen können.

**Comment by Kurpanik** [comment]
**Entscheidung Telko 27.01.**

Die parametrische Geometrie wird nicht embedded, sondern immer als Datei im Container neben der LDF liegen und referenziert
